code/modules/mob/living/simple_animal/hostile/swarmer.dm

```diff
@@ -0,0 +1,834 @@
+		return ..()
+	else
+		return ..()
+		
+/mob/living/simple_animal/hostile/swarmer/MiddleClickOn(atom/A)
+	. = ..()
+	if(dronelist.len == 0)
+		return
+	var/turf/T = get_turf(A)
+	if(!T)
+		return
+	for(var/mob/living/simple_animal/hostile/swarmer/melee/drone in dronelist)
+		drone.LoseTarget()
+		drone.Goto(T, drone.move_to_delay)
+
+/mob/living/simple_animal/hostile/swarmer/CtrlClickOn(atom/A)
+	face_atom(A)
+	if(!isturf(loc))
+		return
+	if(next_move > world.time)
+		return
+	if(!A.Adjacent(src))
+		return
+	A.swarmer_act(src)
+
+/**
+  * Determines what happens to an atom when a swarmer interacts with it
+  *
+  * Determines behavior upon being interacted on by a swarmer.
+  * Arguments:
+  * * S - A reference to the swarmer doing the interaction
+  */
+/atom/proc/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)
+	S.DisIntegrate(src)
+	return TRUE //return TRUE/FALSE whether or not an AI swarmer should try this swarmer_act() again, NOT whether it succeeded.
+
+/turf/closed/indestructible/swarmer_act()
+	return FALSE
+
+/obj/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)
+	if(resistance_flags & INDESTRUCTIBLE)
+		return FALSE
+	for(var/mob/living/L in contents)
+		if(!issilicon(L) && !isbrain(L))
+			to_chat(S, "<span class='warning'>An organism has been detected inside this object. Aborting.</span>")
+			return FALSE
+	return ..()
+
+/obj/item/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)
+	return S.Integrate(src)
+
+/**
+  * Return used to determine how many resources a swarmer gains when consuming an object
+  */
+/atom/movable/proc/IntegrateAmount()
+	return 0
+
+/obj/item/IntegrateAmount() //returns the amount of resources gained when eating this item
+	if(custom_materials)
+		if(custom_materials[SSmaterials.GetMaterialRef(/datum/material/iron)] || custom_materials[SSmaterials.GetMaterialRef(/datum/material/glass)])
+			return 1
+	return ..()
+
+/obj/item/gun/swarmer_act()//Stops you from eating the entire armory
+	return FALSE
+
+/turf/open/swarmer_act()//ex_act() on turf calls it on its contents, this is to prevent attacking mobs by DisIntegrate()'ing the floor
+	return FALSE
+
+/obj/structure/lattice/catwalk/swarmer_catwalk/swarmer_act()
+	return FALSE
+
+/obj/structure/swarmer/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)
+	if(S.AIStatus == AI_ON)
+		return FALSE
+	else
+		return ..()
+
+/obj/effect/swarmer_act()
+	return FALSE
+
+/obj/effect/decal/cleanable/robot_debris/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)
+	S.DisIntegrate(src)
+	qdel(src)
+	return TRUE
+
+/obj/structure/swarmer_beacon/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)
+	to_chat(S, "<span class='warning'>This machine is required for further reproduction of swarmers. Aborting.</span>")
+	return FALSE
+	
+/obj/structure/flora/swarmer_act()
+	return FALSE
+
+/turf/open/lava/swarmer_act()
+	if(!is_safe())
+		new /obj/structure/lattice/catwalk/swarmer_catwalk(src)
+	return FALSE
```

Comment:
Introduce some kind of resource cost, maybe?

Reply:
Considering that these are station antagonists, and the extreme unlikelihood that lava will ever be present in such a scenario, I'm content to leave these to be free.  Pretty sure this line only exists for the unused swarmer on lavaland shtick anyway.